Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm

```diff
@@ -241,34 +247,61 @@ explicit RemoteLayerTreeDisplayLinkClient(WebPageProxyIdentifier pageID)
     return animation;
 }
 
+static RetainPtr<CABasicAnimation> fillFowardsAnimationWithKeyPathAndValue(NSString *keyPath, NSValue *value)
+{
+    RetainPtr animation = fillFowardsAnimationWithKeyPath(keyPath);
+    [animation setFromValue:value];
+    [animation setToValue:value];
+    return animation.autorelease();
+}
+
 static RetainPtr<CABasicAnimation> transientZoomTransformOverrideAnimation(const TransformationMatrix& transform)
 {
-    auto animation = fillFowardsAnimationWithKeyPath(@"transform");
+    return fillFowardsAnimationWithKeyPathAndValue(@"transform", [NSValue valueWithCATransform3D:transform]);
+}
 
-    RetainPtr transformValue = [NSValue valueWithCATransform3D:transform];
-    [animation setFromValue:transformValue.get()];
-    [animation setToValue:transformValue.get()];
+static RetainPtr<CABasicAnimation> transientSizeAnimation(const FloatSize& size)
+{
+    return fillFowardsAnimationWithKeyPathAndValue(@"bounds.size", [NSValue valueWithSize:size]);
+}
 
-    return animation;
+static RetainPtr<CABasicAnimation> transientPositionAnimation(const FloatPoint& position)
+{
+    return fillFowardsAnimationWithKeyPathAndValue(@"position", [NSValue valueWithPoint:position]);
 }
 
 void RemoteLayerTreeDrawingAreaProxyMac::applyTransientZoomToLayer()
 {
     ASSERT(m_transientZoomScale);
-    ASSERT(m_transientZoomOrigin);
+    ASSERT(m_transientZoomOriginInLayerForPageScale);
 
     RetainPtr layerForPageScale = remoteLayerTreeHost().layerForID(m_pageScalingLayerID);
     if (!layerForPageScale)
         return;
 
     TransformationMatrix transform;
-    transform.translate(m_transientZoomOrigin->x(), m_transientZoomOrigin->y());
+    transform.translate(m_transientZoomOriginInLayerForPageScale->x(), m_transientZoomOriginInLayerForPageScale->y());
     transform.scale(*m_transientZoomScale);
 
+    RetainPtr clipLayer = remoteLayerTreeHost().layerForID(m_mainFrameClipLayerID);
+    RetainPtr scrolledContentsLayer = remoteLayerTreeHost().layerForID(m_scrolledContentsLayerID);
+
+    auto scaleForClipLayerAdjustment = std::max(1.0, *m_transientZoomScale);
+    auto clipLayerPosition = FloatPoint { [clipLayer position] };
+    auto clipLayerZoomOrigin = clipLayerPosition + *m_transientZoomOriginInVisibleRect;
```

Comment:
Consider `ASSERT(m_transientZoomOriginInVisibleRect)`

Reply:
Good call, will add this next to the extant `ASSERT(m_transientZoomOrigin);`.

Thanks for the review!